build/helper/metadata_find.py

```diff
@@ -5,11 +5,17 @@ def find_parameter(name, parameters):
     return parameter[0]
 
 
-def find_size_parameter(parameter, parameters):
+def find_size_parameter(parameter_list, parameters):
     '''Returns the parameter that is used to specify the size of another parameter. Applies to 'ivi-dance' and 'passed-in'.'''
-    if not parameter:
+    if len(parameter_list) == 0:
         return None
-    return find_parameter(parameter['size']['value'], parameters)
+    # We verified earlier that all paramers have the same size parameter, so we only need to use the first one
+    try:
+        # Try first as a list
+        return find_parameter(parameter_list[0]['size']['value'], parameters)
+    except KeyError:
+        # Not a list, so must be a single parameter
+        return find_parameter(parameter_list['size']['value'], parameters)
```

Comment:
This is weird... why is parameter_list sometimes a list and sometimes not?

Reply:
Some places we call this function from we only have a single item, others multiple. 

I could have made the change that the places that only have a single parameter to turn it into a list in the function call. `find_size_parameter([param], parameters)` There are more than half a dozen locations that would need this and that seems odd looking.

Instead I chose to handle either in the function. Both solutions would work and I didn't feel either option was better than the other.

Reply:
ok